### ISSUE TITLE
arm64 simulator support

### ISSUE THREAD

**Issue by mvaios** [cleverpush/cleverpush-ios-sdk#78]
Hi, are there any plans to add arm64 simulator support?

**Comment by gbhrdt** [comment]
Hi @mvaios thanks for your suggestion. I just added this to our backlog, we would be happy to add arm64 simulator support soon (probably in 2-3 weeks). I will keep you updated here!

**Comment by mvaios** [comment]
That would be awesome, thank you! 
Looking forward to it.

**Comment by cpandya25** [comment]
@mvaios , Can you please provide error details so we can check accordingly and as well M1 chip

**Comment by lightrow** [comment]
Using Cleverpush in React Native over here, i have to exclude arm64 when building for simulator and run simulator through Rosetta, otherwise it doesn't build

**Comment by mvaios** [comment]
Hi @cpandya25, thanks for getting back to me. I was using cocoapods when opened this issue, but now I switched top SPM, seems like arm64 simulator is working with SPM. With cocoapods the error was that symbols were not defined for arm64 I think, sorry don't have the exact error to send. 
My guess is that it happens because of these 2 lines in .podspec file:
```
s.pod_target_xcconfig     = { 'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => 'arm64' }
s.user_target_xcconfig    = { 'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => 'arm64' }
```

Did you use SPM or CocoaPods for your test? Also keep in mind that only iOS 14 or later simulator is running on arm64, ios13 simulator is still running as intel, even in m1 macs.

**Comment by gbhrdt** [comment]
Hi @lightrow quick question: Did you use Objective-C or Swift when you created your NotificationServiceExtension target? I recommend using Objective-C, otherwise I would suggest to write us to support@cleverpush.com so we can check your individual issue and take a closer look at your implementation. Thanks!

**Comment by lightrow** [comment]
@gbhrdt i used Objective-C

**Comment by lightrow** [comment]
> The proposed answers are outdated/incorrect.
>
>You should initially try to update both [CocoaPods](https://en.wikipedia.org/wiki/CocoaPods) and the dependencies for your library/app, and then, if that doesn't work, contact the vendors of any dependencies you are using to see if they have an update in progress to add support for arm64 Simulator slices on [M1](https://en.wikipedia.org/wiki/Apple_M1) Macs.
>
>There are a lot of answers on here marked as correct suggesting that you should exclude arm64 from the list of supported architectures. This is at best a very temporary workaround, and at worst it will spread this issue to other consumers of your libraries. If you exclude the arm64 Simulator slice, there will be performance impacts on apps that you're developing in the Simulator (which in turn can lead to reduced battery time for your shiny new M1 kit while you're developing your amazing ideas).
>
>https://stackoverflow.com/a/66233165/9313209

**Comment by gbhrdt** [comment]
Hi @lightrow  @mvaios  Thanks for your patience so far! We are currently working on resolving the issue by publishing our library as an XCFramework. We will keep you updated here on the progress.

**Comment by gbhrdt** [comment]
We just released version 1.22.0 of the SDK which brings arm64 simulator support via XCFramework.

@lightrow I will also push a React Native SDK update now so you will be using the latest iOS SDK there as well.

**Comment by mvaios** [comment]
Looks good, thanks!
